x-pack/plugins/reporting/server/lib/content_stream.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { Duplex } from 'stream';
+import type { ElasticsearchClient } from 'src/core/server';
+import { ReportingCore } from '..';
+import { ReportDocument } from '../../common/types';
+
+type Callback = (error?: Error) => void;
+type SearchRequest = Required<Parameters<ElasticsearchClient['search']>>[0];
+
+interface ContentStreamDocument {
+  id: string;
+  index: string;
+  if_primary_term?: number;
+  if_seq_no?: number;
+}
+
+export class ContentStream extends Duplex {
```

Comment:
What is the expected behaviour of writing to a stream? Will we only ever allow writing to non-existing document IDs?